test/helpers/bucket.js

```diff
@@ -0,0 +1,33 @@
+import { GetObjectCommand } from '@aws-sdk/client-s3'
+import pRetry from 'p-retry'
+
+/**
+ * @param {import('@aws-sdk/client-s3').S3Client} client
+ * @param {string} bucketName
+ * @param {string} key
+ */
+export async function waitForBucketItem (client, bucketName, key) {
```

Comment:
I think name is really misleading here, it should probably be called poll as opposed to wait.

```suggestion
export async function pollBucketItem (client, bucketName, key) {
```